webhook/resources/volume/mutator.go

```diff
@@ -260,40 +198,79 @@ func (v *volumeMutator) Create(request *admission.Request, newObj runtime.Object
 		if string(volume.Spec.OfflineReplicaRebuilding) == "" {
 			patchOps = append(patchOps, fmt.Sprintf(`{"op": "replace", "path": "/spec/offlineReplicaRebuilding", "value": "%s"}`, longhorn.OfflineReplicaRebuildingIgnored))
 		}
-	} else {
-		// Always mutate the offlineReplicaRebuilding to disabled for non-SPDK volumes
-		patchOps = append(patchOps, fmt.Sprintf(`{"op": "replace", "path": "/spec/offlineReplicaRebuilding", "value": "%s"}`, longhorn.OfflineReplicaRebuildingDisabled))
 	}
 
+	var patchOpsInCommon admission.PatchOps
+	var err error
+	if patchOpsInCommon, err = mutate(newObj, moreLabels); err != nil {
+		return nil, err
+	}
+	patchOps = append(patchOps, patchOpsInCommon...)
+
 	return patchOps, nil
 }
 
 func (v *volumeMutator) Update(request *admission.Request, oldObj runtime.Object, newObj runtime.Object) (admission.PatchOps, error) {
+	volume := newObj.(*longhorn.Volume)
 	var patchOps admission.PatchOps
 
+	if volume.Spec.AccessMode == "" {
+		patchOps = append(patchOps, `{"op": "replace", "path": "/spec/accessMode", "value": "rwo"}`)
+	}
+
+	size := util.RoundUpSize(volume.Spec.Size)
```

Comment:
Looks like this one is duplicated with https://github.com/longhorn/longhorn-manager/blob/f87d716b8b23b3b0463b588df1a7e4eab34f8f4a/webhook/resources/volume/mutator.go#L221

Reply:
You are right. Since create and update have different logic around this (with create potentially writing a new size for the volume and update simply rounding it up) I have removed this action from the shared logic section in an amend.